elastic/kibana
# [Security Solution][Entity Analytics] Show contributions on risk explainability expanded tab

This PR adds the contribution scores used when calculating an entity's risk score. 
These can be viewed by opening an entity's details flyout and clicking on `View risk contributions` which will open the explainability tab.

The new tab now shows the contribution from different contexts and individual alerts. Currently, there's only one context: Asset Criticality.
The alert inputs shown are the top 10 (at most) alerts used as an input at the time of the scoring calculation. If the final score used more than 10 alerts, we display a message indicating the leftover sum contribution.

This work also updates the server side in order to store the contribution of each individual alert in the risk document itself. We now query the document to retrieve the inputs and then fetch the respective alerts on opening the tab.

Example:
![Screenshot 2024-04-05 at 14 34 24](https://github.com/elastic/kibana/assets/2423976/a4efed46-05cd-4e31-9345-f46472358544)

### How to test

1. Generate some alerts - you can use https://github.com/elastic/security-documents-generator
2. Enable risk scoring via `Security > Manage > Entity Risk Score`  
3. Open the entity details flyout and click `View risk contributions`

Make sure to enable Asset Criticality in `Stack Management > Advanced Settings` if you want to see the criticality contributions.
